src/groovy/org/transmartproject/db/dataquery/highdim/tworegion/JunctionsRow.groovy

```diff
@@ -0,0 +1,33 @@
+package org.transmartproject.db.dataquery.highdim.tworegion
+
+import org.apache.commons.lang.NullArgumentException
+import org.transmartproject.core.dataquery.highdim.AssayColumn
+import org.transmartproject.db.dataquery.highdim.AbstractDataRow
+
+/**
+ * Created by j.hudecek on 23-1-2015.
+ */
+class JunctionsRow extends AbstractDataRow  {
+
+    @Override
+    String getLabel() {
+        def nonnull = data.findAll()
+        if (nonnull.empty)
+            return ""
+        return nonnull.collect({it.getLabel()}).join('; ')
+    }
+
+    public JunctionsRow(Map<AssayColumn, Integer> map ) {
+        if (map == null)
+            throw new NullArgumentException("assayIndexMap can't be null")
+        assayIndexMap = map
+        data = new ArrayList(map.size())
+    }
+
+    public void addJunction(DeTwoRegionJunction r) {
+        data[assayIndexMap.find({it.key.id == r.assay.id}).value] = r.toMap()
+        allJunctions.add(r)
+    }
+
+    Set<DeTwoRegionJunction> allJunctions = new HashSet<DeTwoRegionJunction>()
+}
```

Comment:
It's still not clear to me what a `JunctionsRow` aggregates. A comment on the class would be helpful. The duplication of data is also confusing and rather unfortunate. Do away with `allJunctions` and `DeTwoRegionJunction::toMap()` and store the junction directly in `data` (even if it implies fixing the alldatatests, which is testing too much right now IIRC).